File: engine/engine-model/src/main/scala/tools/packet/event/HitSnowBall.scala
package tools.packet.event

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class HitSnowBall(private var _what: Int, private var _damage: Int) extends PacketInput {
  def what: Int = _what

  def damage: Int = _damage

  override def opcode(): SendOpcode = SendOpcode.HIT_SNOWBALL
}

File: engine/engine-model/src/main/java/tools/packet/event/HitSnowBall.java
package tools.packet.event;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record HitSnowBall(Integer what, Integer damage) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.HIT_SNOWBALL;
   }
}